### PR TITLE
Implement pinned messages + message deletion

### LINKED ISSUE #17: Pinned messages
Just like Discord - but without the 50 pin limit.

- [x] Pin messages API
- [ ] Unpin messages API
- [x] List pin messages API

### PR DESCRIPTION
**Pull request by heyitsmeuralex**
@PullJosh did a design for tabs in the right sidebar; this is it in fruition!

* Contains some API spec changes: two new events and an unpin endpoint.
* You can delete/pin/unpin messages now with the UI (see #115, fixes #17)
* Note that the `mentions` tab is left blank for when #92 is finally complete, and the edit message icon does nothing when clicked; see #115.
* Includes / iterates on #254 - channels are now loaded from the oldest unread message
* ~~Blocked by / includes #253. Merge that first.~~ :ballot_box_with_check:

### CODE REVIEW

**Comment by towerofnix** [comment]
**Pre-code-review notes:**

Note - trying to view the "pins" tab while no channel is selected causes an error, because it fetches `/api/channels/undefined/pins`. Pins tab should probably be hidden while not actively in a channel? (And should switch to "users" tab when going back to a state with no channel active.)

Note - it's sorta hard to tell exactly what message my mouse is on top of, since the icons appear so far to the right. Maybe a dotted line from the message icons to the first line of the related message would help?

Note - on an authorized server, when I reloaded the page, there was an "authorization related error", but I'm 60% sure that doesn't have anything to do with this.

Note - flairs don't show up on pinned messages. I assume this is intentional? (It looks better without flairs imo.)

Note - the most recently pinned messages show up at the bottom of the pinned message list. IMO there should be a way to sort pinned messages - maybe even manually. Otherwise, they should probably be sorted by date. (Should the newest messages go on the top or the bottom?)

Note - the highlight from jumping to a pinned message doesn't look any different in the dark mode, but this is probably fine.

File: docs/api.md [hunk 1]
```diff
@@ -475,18 +475,27 @@ Model:
...
+  "unreadMessageCount": number,
+
+  "oldestUnreadMessageID": ID | null,
 }
 ```
 
 Related events:
 * [channel/new](#channel-new)
 * [channel/update](#channel-update)
+* [channel/pins-update](#channel-pins-update)
```
Comment (towerofnix) [inline docs/api.md]: This event does not exist.

File: docs/api.md [hunk 2]
```diff
@@ -661,7 +670,7 @@ GET /api/channels/5678/pins
 + **in-url** id (ID)
 + `messageID` (ID) - The message to pin to this channel.
 
-Returns `{}` if successful.
+Returns `{}` if successful. Emits [channel/pins-update](#channel-pins-update).
```
Comment (towerofnix) [inline docs/api.md]: This should be `channel/pins/add`.

File: packages/server/serialize.js [hunk 3]
```diff
@@ -52,6 +52,15 @@ module.exports = function makeSerializers({util, db}) {
 
       if (sessionUser) {
         obj.unreadMessageCount = await getUnreadMessageCountInChannel(sessionUser, c._id)
+
+        if (obj.unreadMessageCount === 0) {
+          obj.oldestUnreadMessageID = null
+        } else {
+          const msg = await getOldestUnreadMessageInChannel(sessionUser, c._id)
+
+          console.log(msg)
```
Comment (towerofnix) [inline packages/server/serialize.js]: Leftover console.log?

File: docs/api.md (hunk 1 above)
Comment (towerofnix) [reply docs/api.md]: (Replace this with channel/pins/add and channel/pins/remove).

**Comment by heyitsmeuralex** [comment]
Hold on, just adding that dotted line thing you asked for.

**Pull request merged**